packer/linux_debian.pkr.hcl

```diff
@@ -131,6 +131,10 @@ build {
     inline = [
       <<-SCRIPT
         apt-get update -y
+
+        # first update grub2-common, otherwise grub-cloud-amd64 fails
+        DEBIAN_FRONTEND=noninteractive apt-get upgrade grub2-common -y
+
```

Comment:
Shouldn't be needed anymore, right?

Reply:
Unfortunately, no. If I remove this part, CI fails with:

```
linux.googlecompute.sid: Setting up linux-image-6.5.0-2-cloud-amd64 (6.5.6-1) ...
linux.googlecompute.sid: /etc/kernel/postinst.d/initramfs-tools:
linux.googlecompute.sid: update-initramfs: Generating /boot/initrd.img-6.5.0-2-cloud-amd64
linux.googlecompute.sid: /etc/kernel/postinst.d/zz-update-grub:
linux.googlecompute.sid: Generating grub configuration file ...
linux.googlecompute.sid: /etc/grub.d/10_linux: 1: version_find_latest: not found
linux.googlecompute.sid: run-parts: /etc/kernel/postinst.d/zz-update-grub exited with return code 127
linux.googlecompute.sid: dpkg: error processing package linux-image-6.5.0-2-cloud-amd64 (--configure):
linux.googlecompute.sid:  installed linux-image-6.5.0-2-cloud-amd64 package post-installation script subprocess returned error exit status 1
```

Reply:
That's odd. Clearly some broken dependencies...

Reply:
Something seems to have lead to debian sid to fail to build with this PR, even though it's succeeding on main currently:
https://cirrus-ci.com/build/5861999236612096

I suspect it's the change above. Just upgrading grub2-common fails because its dependencies aren't ugraded, and because you're using upgrade, not dist-upgrade, it can't upgrade those, because it'd entail more work than upgrade does.

Reply:
Another thing: Why would this change affect debian sid and thus potentially require adding the above line? Shouldn't the sid build be unaffected?